File: problem5/smallestMultiple.java
import java.util.*; // for ArrayList

/**
* @author Claire Jaja
* @version 1/13/14
*
* Project Euler Problem 5
* Smallest multiple
* 2520 is the smallest number that can be divided by each of the numbers from 1 to 10 without any remainder.
* What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20?
*/

public class smallestMultiple
{
	public static void main(String[] args) {
        boolean smallestMultipleFound = false;
        long currentNumber = 20;

        while (!smallestMultipleFound) {
            if (isEvenlyDivisibleByAllNumbersToTwenty(currentNumber)) {
                smallestMultipleFound = true;
                System.out.println(currentNumber);
            } else {
                currentNumber++;
            }
        }
    }

    public static boolean isEvenlyDivisibleByAllNumbersToTwenty(long num) {
       for (int i = 20; i >= 1; i--) {
            if (num % i != 0) {
                return false;
            }
        }
        return true; 
    }
}

File: src/python/problem5/smallest_multiple.py
# Claire Jaja
# 11/4/2014
#
# Project Euler
# Problem 5
# Smallest multiple
#
# 2520 is the smallest number that can be divided by each of the numbers
# from 1 to 10 without any remainder.
# What is the smalelst positive number that is evenly divisible
# by all the numbers from 1 to 20?

def main():
    smallest_multiple_found = False
    current_num = 2520
    while not smallest_multiple_found:
        if is_evenly_divisible_to_twenty(current_num):
            smallest_multiple_found = True
            print(current_num)
        else:
            current_num += 20

def is_evenly_divisible_to_twenty(x):
    for i in range(20,0,-1):
        if x % i != 0:
            return False
    return True

if __name__ == "__main__":
    main()
